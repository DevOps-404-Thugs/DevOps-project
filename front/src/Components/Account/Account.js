import React, { useEffect, useState } from 'react'
import Axios from 'axios'
import { Link } from 'react-router-dom';
import Card from 'react-bootstrap/Card';


function Account() {

    const [user,setUser] = useState({username:null, password:null})

    useEffect(() => {
        Axios.get(`/account`)
            .then(response => {
                console.log(response.data);
                setUser(response.data);
            })
            .catch(function(error){
                console.log(error.response)
                
            })
        
        console.log(user);    
    });
		

    return (
        <div style={{ width: '50%', margin: '3rem auto'}}>
            {user.username == null?
                <h2 style={{ display: 'flex', justifyContent: 'center' }}>You need to login first!</h2>
                :
                <div>
                    <Card style={{ 'margin-bottom': '30px' }}>
                        <Card.Header style={{ textAlign: 'center' }}>Account Info</Card.Header>
                        <Card.Title style={{ 'padding-top': '0.75rem','padding-left': '1rem'}}>username: {user.username}</Card.Title>
                        <Card.Text style={{ 'padding-bottom': '0.75rem','padding-left': '1rem', 'font-weight':'bold'}}>email: {user.email}</Card.Text>
                    </Card>
                    <Link className="link" to="/account_update" style={{color:'blue',display: 'flex', justifyContent: 'center'}}>Update Account Information</Link>
                </div>
			}
            

            <br /><br />


            <div style={{ display: 'flex', justifyContent: 'center' }}>
                
            </div>

        </div>
    )
}

export default Account;