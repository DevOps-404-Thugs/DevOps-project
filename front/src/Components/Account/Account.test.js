import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Account from './Account';


describe('Account component', () =>{
    test("match the snapshot", () =>{
        const component = renderer.create(
            <BrowserRouter>
                <Account/>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });  
});