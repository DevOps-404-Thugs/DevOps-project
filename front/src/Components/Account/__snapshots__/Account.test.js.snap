// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component match the snapshot 1`] = `
<div
  style={
    Object {
      "margin": "3rem auto",
      "width": "50%",
    }
  }
>
  <h2
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    You need to login first!
  </h2>
  <br />
  <br />
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  />
</div>
`;
